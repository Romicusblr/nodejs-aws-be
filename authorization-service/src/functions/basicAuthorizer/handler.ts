import "source-map-support/register";

import { middyfy } from "@libs/lambda";
import {
  APIGatewayAuthorizerResult,
  APIGatewayTokenAuthorizerHandler,
} from "aws-lambda";

const { USERNAME, PASSWORD } = process.env;
import schema from "./schema";

const handler: APIGatewayTokenAuthorizerHandler = async (event) => {
  const { authorizationToken, methodArn } = event;
  let effect = "Deny";
  let principalId;
  try {
    const credsB64 = authorizationToken.split(" ")[1];
    const [user, pass] = Buffer.from(credsB64, "base64")
      .toString("utf-8")
      .split(":");

    console.dir({ user, pass, USERNAME, PASSWORD });
    principalId = user;

    if (user === USERNAME && pass === PASSWORD) {
      effect = "Allow";
      console.dir({ effect });
    }
  } catch (err) {
    console.log(err);
  }
  const policy = generatePolicy(principalId, effect, methodArn);
  return policy;
};

function generatePolicy(user, effect, resource): APIGatewayAuthorizerResult {
  return {
    principalId: user,
    policyDocument: {
      Version: "2012-10-17",
      Statement: [
        {
          Effect: effect,
          Action: "execute-api:Invoke",
          Resource: resource,
        },
      ],
    },
  };
}

export const main = middyfy(handler, schema);
