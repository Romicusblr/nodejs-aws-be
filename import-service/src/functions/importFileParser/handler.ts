import "source-map-support/register";
import csv from "csv-parser";
import { middyfy } from "@libs/lambda";
import {
  S3Client,
  GetObjectCommand,
  CopyObjectCommand,
  CopyObjectCommandInput,
  DeleteObjectCommand,
} from "@aws-sdk/client-s3";
import type { S3Event, Handler } from "aws-lambda";
import config from "@libs/config";
import schema from "./schema";

const client = new S3Client({});

function getParsedKey(key) {
  return key.replace(config.app.uploadFolder, config.app.parsedFolder);
}

const handler: Handler<S3Event> = async (event) => {
  try {
    for (const record of event?.Records) {
      const bucket = record.s3.bucket.name;
      const key = record.s3.object.key;
      const action = record.eventName;
      console.dir({ bucket, key, action });

      const params = {
        Bucket: bucket,
        Key: key,
      };

      const res = await client.send(new GetObjectCommand(params));

      await new Promise((resolve, reject) => {
        res.Body.pipe(csv())
          .on("data", (data) => console.log(data))
          .on("error", reject)
          .on("end", resolve);
      });

      const copyParams: CopyObjectCommandInput = {
        Bucket: bucket,
        Key: getParsedKey(key),
        CopySource: `${bucket}/${key}`,
      };

      await client.send(new CopyObjectCommand(copyParams));

      await client.send(new DeleteObjectCommand(params));
    }
    return true;
  } catch (err) {
    console.log(err);
    return false;
  }
};

export const main = middyfy(handler, schema);
