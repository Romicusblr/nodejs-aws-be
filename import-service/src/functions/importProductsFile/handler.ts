import "source-map-support/register";

import type { ValidatedEventAPIGatewayProxyEvent } from "@libs/apiGateway";
import { formatJSONResponse } from "@libs/apiGateway";
import { middyfy } from "@libs/lambda";
import createError from "http-errors";

import schema from "./schema";

const handler: ValidatedEventAPIGatewayProxyEvent<typeof schema> = async (event) => {
  try {
    const name = event?.queryStringParameters?.name;
    return formatJSONResponse({  });
  } catch (err) {
    console.log(err);
    return formatJSONResponse(createError(500));
  }
};

export const main = middyfy(handler, schema);
