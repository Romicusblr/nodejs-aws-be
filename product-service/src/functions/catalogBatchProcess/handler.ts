import "source-map-support/register";

import type { SQSEvent, Handler } from "aws-lambda";
// import { middyfy } from "@libs/lambda";
import DBTransactionDAO from "@daos/db_transaction";

// import schema from "./schema";

const handler: Handler<SQSEvent> = async (event) => {
  let dao: DBTransactionDAO;
  try {
    dao = new DBTransactionDAO();
    for (const record of event.Records) {
      console.dir({ record });
    }
    // const productList = await dao.createProduct(product);
    return true;
  } catch (err) {
    console.log(err);
    return false;
  } finally {
    dao.client.destroy();
  }
};

export const main = handler;
