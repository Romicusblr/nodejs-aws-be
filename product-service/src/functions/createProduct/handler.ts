import "source-map-support/register";

import type { ValidatedEventAPIGatewayProxyEvent } from "@libs/apiGateway";
import { formatJSONResponse } from "@libs/apiGateway";
import { middyfy } from "@libs/lambda";
import DBTransactionDAO from "@daos/db_transaction";
import createError from "http-errors";

import schema from "./schema";

const handler: ValidatedEventAPIGatewayProxyEvent<typeof schema> = async (event) => {
  let dao: DBTransactionDAO;
  try {
    dao = new DBTransactionDAO();
    const product = event.body;
    const productList = await dao.createProduct(product);
    return formatJSONResponse({ productList });
  } catch (err) {
    console.log(err);
    return formatJSONResponse(createError(500));
  } finally {
    dao.client.destroy();
  }
};

export const main = middyfy(handler, schema);
