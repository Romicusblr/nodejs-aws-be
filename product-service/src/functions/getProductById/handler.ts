import "source-map-support/register";

import type { ValidatedEventAPIGatewayProxyEvent } from "@libs/apiGateway";
import { formatJSONResponse } from "@libs/apiGateway";
import { middyfy } from "@libs/lambda";

import schema from "./schema";
import productList from "../../productList.json";

const handler: ValidatedEventAPIGatewayProxyEvent<typeof schema> = async (
  event
) => {
  const productId = event?.pathParameters?.productId;

  const product = productList.find(({ id }) => id === productId);

  if (!product) {
    throw new Error(`Product with id ${productId} is not found`);
  }

  return formatJSONResponse({ product });
};

export const main = middyfy(handler);
