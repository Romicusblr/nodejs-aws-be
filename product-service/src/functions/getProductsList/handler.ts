import "source-map-support/register";

import type { ValidatedEventAPIGatewayProxyEvent } from "@libs/apiGateway";
import { formatJSONResponse } from "@libs/apiGateway";
import { middyfy } from "@libs/lambda";
import ProductDAO from "@daos/product";
import createError from "http-errors";

import schema from "./schema";

const handler: ValidatedEventAPIGatewayProxyEvent<typeof schema> = async () => {
  let productDAO: ProductDAO;
  try {
    productDAO = new ProductDAO();
    const productList = await productDAO.searchWithCount({});
    return formatJSONResponse({ productList });
  } catch (err) {
    return formatJSONResponse(createError(500, err.message));
  } finally {
    productDAO.client.destroy();
  }
};

export const main = middyfy(handler);
