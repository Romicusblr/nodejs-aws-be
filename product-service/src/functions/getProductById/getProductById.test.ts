//@ts-nocheck
import {main as getProductById} from "./handler";

describe("getProductById function", () => {
  test("returns correct object", async () => {
    const res = await getProductById({ pathParameters: { productId: "1" } });
    const product = JSON.parse(res.body);
    expect(product).toMatchSnapshot();
  });

  test("throws error for wrong id", async () => {
    await expect(
      getProductById({ pathParameters: { productId: "10" } })
    ).rejects.toThrow();
  });
});
