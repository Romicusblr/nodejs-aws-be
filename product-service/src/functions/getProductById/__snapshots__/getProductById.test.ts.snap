// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getProductById function returns correct object 1`] = `
Object {
  "message": Object {
    "product": Object {
      "count": 4,
      "description": "With a 1-inch sensor, 5.4K video, MasterShots, and much more, DJI Air 2S is the ultimate all-in-one camera drone.",
      "id": "1",
      "price": 1029,
      "title": "DJI Air 2S",
    },
  },
}
`;
